tfprotov5/internal/tf5serverlogging/client_capabilities.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tf5serverlogging
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-go/internal/logging"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+)
+
+// ConfigureProviderClientCapabilities generates a TRACE "Announced client capabilities" log.
+func ConfigureProviderClientCapabilities(ctx context.Context, capabilities *tfprotov5.ConfigureProviderClientCapabilities) {
+	responseFields := map[string]interface{}{
+		logging.KeyClientCapabilityDeferralAllowed: false,
+	}
+
+	if capabilities != nil {
+		responseFields[logging.KeyClientCapabilityDeferralAllowed] = capabilities.DeferralAllowed
+	}
+
+	logging.ProtocolTrace(ctx, "Announced client capabilities", responseFields)
+}
```

Comment:
I think this setup is reasonable to show the TRACE log even with older versions of Terraform that wouldn't have this client capability -- otherwise developers/maintainers will need to pay attention to the omission of the log entry instead. Another way to do this logging would be to key a different "No announced client capabilities" log message when `capabilities != nil`, but the current setup is also fine in my book. 👍

Reply:
Does f029631 match what you were thinking?

Reply:
f02963174f74fb70d8d39c47b95f94adeade54e0 does indeed match! Thank you.